### PR TITLE
Support reading VCF files from Hadoop.

### PR DESCRIPTION
**Pull request by tomwhite**

### CODE REVIEW

**Comment by tomwhite** [comment]
This is ready for review, @davidadamsphd, @lbergelson.

File: src/main/java/org/broadinstitute/hellbender/utils/variant/Variant.java [hunk 1]
```diff
@@ -10,7 +11,7 @@
  * NOTE: getStart() and getEnd() are 1-base inclusive (which matches the current GATK tools).
  * This does not match the GA4GH spec.
  */
-public interface Variant extends Locatable {
+public interface Variant extends Locatable, Serializable {
```
Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/utils/variant/Variant.java]: Why is this necessary? We already have a coder for the Variant interface.

**Comment by davidadamsphd** [comment]
Sorry for the delay, I was OOO Wed-Friday last week, then had a visitor from another team in town for the last two days.

Just the one question. Otherwise looks great.

File: src/main/java/org/broadinstitute/hellbender/utils/variant/Variant.java (hunk 1 above)
Comment (tomwhite) [inline src/main/java/org/broadinstitute/hellbender/utils/variant/Variant.java]: You're right - I should just set the coder. New version coming up.

**Comment by akiezun** [comment]
@davidadamsphd what should we do with this PR? migrate to https://github.com/broadinstitute/hellbender-dataflow? Adapt to spark? both?

**Comment by akiezun** [comment]
@tomwhite is this ready to merge?

**Comment by davidadamsphd** [comment]
@tomwhite, I think migrate it to hellbender-dataflow. We already have something similar in Spark (that I wrote using this as a template).

**Comment by tomwhite** [comment]
Yes, this can go in. I've just rebased it. Can be merged when tests pass.

**Comment by davidadamsphd** [comment]
This broke some tests... not quite sure why.

**Comment by akiezun** [comment]
seems like it may be legit errors, log is in here: https://s3.amazonaws.com/archive.travis-ci.org/jobs/83225736/log.txt

**Comment by davidadamsphd** [comment]
@tomwhite, any luck fixing these tests?

**Comment by tomwhite** [comment]
No, the failures seem to be unrelated to the changes, which is odd. Also, I can't reproduce locally. Rebasing and running again to see if there's any change.

**Comment by tomwhite** [comment]
I'm still at a loss on this one. If I run all tests without this change, they all pass. With this change there are failures in AddContextDataToReadSparkUnitTest, JoinReadsWithRefBasesSparkUnitTest, JoinReadsWithVariantsSparkUnitTest, ReadsSparkSinkUnitTest, but if I run them individually they pass (with the change).

This should probably be moved to hellbender-dataflow anyway, if it's still needed.

**Comment by davidadamsphd** [comment]
@droazen, any ideas?

**Comment by akiezun** [comment]
@davidadamsphd who's migrating this?

**Comment by davidadamsphd** [comment]
I think @tomwhite is (reassigning).

**Comment by lbergelson** [comment]
Closing this since I think we're never doing it.  Re-openning on gatk-dataflow for long term storage
